app/src/ui/branches/branch-list.tsx

```diff
@@ -81,6 +81,9 @@ interface IBranchListProps {
   /** Callback to fire when the filter text is changed */
   readonly onFilterTextChanged: (filterText: string) => void
 
+  /** Callback to fire when the filter text is changed */
+  readonly onEnterClicked?: (text: string) => void
```

Comment:
I don't think we need to expose this: `BranchList` already has a method to create a branch: https://github.com/desktop/desktop/blob/28a2ee45a220dc87592e5c8e4666c4a6006b4f2b/app/src/ui/branches/branch-list.tsx#L281-L285